### PR TITLE
add elemMatch projection operator support to fields parsing

### PR DESCRIPTION
**Pull request by seangarner**
Solution should leave the door open for other projection operators (e.g. `$`, `$slice`).

### CODE REVIEW

File: test/fields.js [hunk 1]
```diff
@@ -73,4 +73,59 @@ describe('fields', function() {
...
+          school:{
+            $eq:102
+          }
+        }
+      }
+    };
+    expect(mongoUrl.fields('-name,elemMatch(students,eq(school,102))')).to.eql(expected);
+  });
+
+  it.skip('should mix elemMatch and exclusive projection when elemMatch is first', function () {
```
Comment (seangarner) [inline test/fields.js:117]: I wish I could get this working, but I'm not convinced it's worth the effort.

File: src/fields.pegjs [hunk 2]
```diff
@@ -31,6 +39,220 @@ including
...
+
+//TODO: make this completely mongo compatible
+Property "document property"
+  = property:$([^\.$,\0\ ][^,\0\ ]*) { return property; }
+
+Scalar "scalar value"
+  = String
+  / Number
+  / Boolean
+  / "null" __  { return null;  }
```
Comment (davidgwking) [inline src/fields.pegjs:181]: Did you leave out the new Date type for a particular reason?

**Comment by davidgwking** [comment]
This is a lot of copy-paste. It's a shame that peg doesn't let you import productions from other files.

It looks like this project may need a more robust build process that can merge source files.

File: src/fields.pegjs (hunk 2 above)
Comment (seangarner) [inline src/fields.pegjs:181]: Thanks for the catch.  I copy/pasted this before that branch was merged.

**Comment by seangarner** [comment]
Yeah, there is certainly some copy paste here... but it's not absolutely clear cut either, because some definitions in query had to be adapted for field parsing.  e.g. there's no `assertCan` method in field parsing so that has to be removed.

But it's something I'm interested in looking into in another branch.

File: src/fields.pegjs (hunk 2 above)
Comment (seangarner) [inline src/fields.pegjs:181]: Fixed in 9e8c65c

**Pull request merged**